test/e2e/specs/fakerp/keyrotation.go

```diff
@@ -38,8 +39,17 @@ var _ = Describe("Key Rotation E2E tests [KeyRotation][Fake][LongRunning]", func
 		ctx = context.WithValue(ctx, api.ContextKeyClientSecret, os.Getenv("AZURE_CLIENT_SECRET"))
 		ctx = context.WithValue(ctx, api.ContextKeyTenantID, os.Getenv("AZURE_TENANT_ID"))
 
+		logrus.SetLevel(logrus.DebugLevel)
+		logrus.SetFormatter(&logrus.TextFormatter{FullTimestamp: true})
+		logrus.SetOutput(GinkgoWriter)
+		log := logrus.NewEntry(logrus.StandardLogger())
+
+		frpconfig, err := fakerp.NewConfig(log, true)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(frpconfig).NotTo(BeNil())
+
 		By("Parsing the external manifest")
-		external, err := parseExternalConfig(*manifest)
+		external, err := fakerp.LoadClusterConfigFromManifest(log, *manifest, frpconfig)
```

Comment:
Please rework `LoadClusterConfigFromManifest` to accept conf.Manifest instead of requiring the whole struct.

Reply:
Also `LoadClusterConfigFromManifest` is very weird, I hope we can simplify it.